File: tests/com/github/redfoos/logstash/parser/LogstashParserTest.java
package com.github.redfoos.logstash.parser;

import com.github.redfoos.logstash.LogstashParserDefinition;
import com.intellij.testFramework.ParsingTestCase;

public class LogstashParserTest extends ParsingTestCase {
    public LogstashParserTest() {
        super("", "conf", new LogstashParserDefinition());
    }

    @Override
    protected String getTestDataPath() {
        return "testData/parser";
    }

    @Override
    protected boolean skipSpaces() {
        return false;
    }

    @Override
    protected boolean includeRanges() {
        return true;
    }

    public void testEmptyPluginBlock() {
        doTest(true);
    }

    public void testPluginBlock() {
        doTest(true);
    }

    public void testConditions() {
        doTest(true);
    }

    public void testDataTypes() {
        doTest(true);
    }

    public void testComplexConfig() {
        doTest(true);
    }

    public void testFieldAccess() { doTest(true);}

}

File: src/test/kotlin/com/github/redfoos/logstash/parser/LogstashParserTest.kt
package com.github.redfoos.logstash.parser

import com.github.redfoos.logstash.LogstashParserDefinition
import com.intellij.testFramework.ParsingTestCase
import org.junit.Test

class LogstashParserTest : ParsingTestCase("", "conf", LogstashParserDefinition()) {
    override fun getTestDataPath(): String {
        return "src/test/testData/parser"
    }

    override fun skipSpaces(): Boolean {
        return false
    }

    override fun includeRanges(): Boolean {
        return true
    }

    @Test
    fun testEmptyPluginBlock() {
        doTest(true)
    }

    @Test
    fun testPluginBlock() {
        doTest(true)
    }

    @Test
    fun testConditions() {
        doTest(true)
    }

    @Test
    fun testDataTypes() {
        doTest(true)
    }

    @Test
    fun testComplexConfig() {
        doTest(true)
    }

    @Test
    fun testFieldAccess() {
        doTest(true)
    }
}
